Assets/SEE/GraphProviders/VCSGraphProvider.cs

```diff
@@ -0,0 +1,421 @@
+        /// </summary>
+        /// <param name="graph">The graph into which the metrics shall be loaded</param>
+        /// <param name="city">This parameter is currently ignored.</param>
+        /// <param name="changePercentage">This parameter is currently ignored.</param>
+        /// <param name="token">This parameter is currently ignored.</param>
+        public override async UniTask<Graph> ProvideAsync(Graph graph, AbstractSEECity city, Action<float> changePercentage = null,
+                                                          CancellationToken token = default)
+        {
+            CheckArguments(city);
+            return await UniTask.FromResult<Graph>(GetVCSGraph(PathGlobbing, RepositoryPath.Path, CommitID));
+        }
+
+        /// <summary>
+        /// Checks whether the assumptions on <see cref="RepositoryPath"/> and
+        /// <see cref="CommitID"/> and <paramref name="city"/> hold.
+        /// If not, exceptions are thrown accordingly.
+        /// </summary>
+        /// <param name="city">To be checked</param>
+        /// <exception cref="ArgumentException">thrown in case <see cref="RepositoryPath"/>,
+        /// or <see cref="CommitID"/>
+        /// is undefined or does not exist or <paramref name="city"/> is null</exception>
+        protected void CheckArguments(AbstractSEECity city)
+        {
+            if (string.IsNullOrEmpty(RepositoryPath.Path))
+            {
+                throw new ArgumentException("Empty repository path.\n");
+            }
+            if (!Directory.Exists(RepositoryPath.Path))
+            {
+                throw new ArgumentException($"Directory {RepositoryPath.Path} does not exist.\n");
+            }
+            if (string.IsNullOrEmpty(CommitID))
+            {
+                throw new ArgumentException("Empty newCommitID.\n");
+            }
+            if (city == null)
+            {
+                throw new ArgumentException("The given city is null.\n");
+            }
+        }
+
+        /// <summary>
+        /// Builds the VCS graph with specific metrics.
+        /// </summary>
+        /// <param name="pathGlobbing">The paths which get included/excluded.</param>
+        /// <param name="repositoryPath">The path to the repository.</param>
+        /// <param name="commitID">The commitID where the files exist.</param>
+        /// <returns>the graph.</returns>
+        static Graph GetVCSGraph(Dictionary<string, bool> pathGlobbing, string repositoryPath, string commitID)
+        {
+            string[] pathSegments = repositoryPath.Split(Path.DirectorySeparatorChar);
+
+            Graph graph = new(repositoryPath, pathSegments[^1]);
+
+            // The main directory.
+            GraphUtils.NewNode(graph, pathSegments[^1], "directory", pathSegments[^1]);
+
+            IEnumerable<string> includedFiles = pathGlobbing
+                .Where(path => path.Value == true)
+                .Select(path => path.Key);
```

Comment:
Make sure to consume this enumerable if it's used more than once, otherwise it is enumerated every time it is accessed. In this case specifically, I recommend using `.ToHashSet` to get $O(1)$ `Contains()`. Also applies to excludedFiles.